secret.go

```diff
@@ -11,42 +17,93 @@ import (
 func applySecret(ctx *cli.Context) error {
 	baseURL := getNGBaseURL(ctx)
 	gitPat := ctx.String("token")
+	filePath := ctx.String("file")
 	gitPat = getGitSecret(gitPat)
+	secretType := ctx.String("secret-type")
+	requiresFile := isFileTypeSecret(secretType)
+	secretIdentifier := getSecretIdentifier(secretType)
+	secretName := getSecretName(secretType)
+	var secretBody HarnessSecret
+	var headers map[string]string
+	var err error
 
-	if gitPat == "" {
+	if requiresFile && filePath == "" {
+		println(fmt.Sprintf("Secret type %s requires file path to create a valid filetype secret", secretType))
+		return nil
+	}
+	if !requiresFile && gitPat == "" {
 		println("Secret cannot be an empty string")
 		return nil
 	}
-	createSecretURL := GetUrlWithQueryParams("", baseURL, "v2/secrets", map[string]string{
+	if requiresFile {
+		headers = make(map[string]string)
+		headers["Content-Type"] = "multipart/form-data"
+	}
+
+	createUrl := "v2/secrets"
+	if requiresFile {
+		createUrl = fmt.Sprintf("v2/secrets/%s", "files")
+	}
+	createSecretURL := GetUrlWithQueryParams("", baseURL, createUrl, map[string]string{
 		"accountIdentifier": cliCdRequestData.Account,
 		"projectIdentifier": DEFAULT_PROJECT,
 		"orgIdentifier":     DEFAULT_ORG,
 	})
-	updateSecretURL := GetUrlWithQueryParams("", baseURL, fmt.Sprintf("v2/secrets/%s", GITHUB_SECRET_IDENTIFIER), map[string]string{
+
+	updateUrl := fmt.Sprintf("v2/secrets/%s", secretIdentifier)
```

Comment:
Make use of the newly declared endpoint as the above comment for better code-maintenance